src/d3a/models/market/market_structures.py

```diff
@@ -252,8 +270,20 @@ def trade_from_JSON_string(trade_string, current_time):
         trade_dict['residual'] = offer_or_bid_from_JSON_string(trade_dict['residual'],
                                                                current_time)
     trade_dict['time'] = parse(trade_dict['time'])
-    # if 'offer_bid_trade_info' in trade_dict:
-    #     trade_dict['offer_bid_trade_info'] = TradeBidInfo(*trade_dict['offer_bid_trade_info'])
+    if 'offer_bid_trade_info' in trade_dict and trade_dict['offer_bid_trade_info'] is not None:
```

Comment:
```suggestion
    if key_in_dict_and_not_none(trade_dict, 'offer_bid_trade_info'):
```
could import `key_in_dict_and_not_none` from `d3a_interface.utils`